### PR TITLE
also install `easybuild` package for `EasyBuild >= 5`

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by Micket — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@branfosj Can you share some more information on this, explain why this is required?

File: easybuild/easyblocks/e/easybuildmeta.py [hunk 1]
```diff
@@ -65,6 +65,9 @@ def __init__(self, *args, **kwargs):
             self.easybuild_pkgs.extend(['vsc-base', 'vsc-install'])
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
+        elif LooseVersion(self.version) >= LooseVersion('5.0'):
+            # use easybuild-base for easybuild to avoid matching all easybuild-* directories during install
+            self.easybuild_pkgs.append('easybuild-base')
```
Comment (jfgrimm) [inline easybuild/easyblocks/e/easybuildmeta.py:70]: what if we guard this to also check that the `easybuild-{version}.tar.gz` package is in the sources? Then in theory this could be merged without https://github.com/easybuilders/easybuild-easyconfigs/pull/22774 for now?
```suggestion
        elif LooseVersion(self.version) >= LooseVersion('5.0') and any(f'easybuild-{self.version}' in source['filename'] for source in self.cfg['sources']):
            # use easybuild-base for easybuild to avoid matching all easybuild-* directories during install
            self.easybuild_pkgs.append('easybuild-base')
```

Comment (branfosj) [reply easybuild/easyblocks/e/easybuildmeta.py:70]: The issue I am expecting is when the new source appears in the easyconfig it will break because someone mayl be using an old easybuild that does not have whatever protection we add in here.

**Comment by branfosj** [comment]
> > I have a concern that we will hit an issue that [easybuilders/easybuild-easyconfigs#22774](https://github.com/easybuilders/easybuild-easyconfigs/pull/22774) will break without the changes in this PR, meaning that the EB install latest command line option will not work. I am not sure how to check that though. This may mean that an alternative solution (so that `easybuild` appears in `pip list`) would be better than the changes here.
> 
> ah, I think I see what you mean. That's not really something we want to break :/ I think creating even an empty `lib/python{ver}/easybuild-5.0.0-py{ver}.egg-info/PKG-INFO` should be sufficient for it to show up in `pip list`?

See #3725

File: easybuild/easyblocks/e/easybuildmeta.py (hunk 1 above)
Comment (jfgrimm) [reply easybuild/easyblocks/e/easybuildmeta.py:70]: indeed; with the addition of the suggestion we could
* merge this as is
* then merge https://github.com/easybuilders/easybuild-easyconfigs/pull/22774 (or equivalent) at some future date (EB 6.0?) -- once this code has been in a few releases it would be less of an issue

that would mean manually generating the egg file would be a temporary workaround

**Comment by Micket** [comment]
Solved in #3725 instead

**Comment by jfgrimm** [comment]
LGTM

**Comment by boegel** [comment]
Don't merge this yet please, I'm concerned this may cause trouble, for example with exsiting `EasyBuild-5.*eb` easyconfigs

**Comment by jfgrimm** [comment]
@boegel was planning to wait until submitting a test report anyway

**Comment by jfgrimm** [comment]
successful report using the `EasyBuild-5.0.0.eb` easyconfig (with `easybuild` added to the sources) in: https://github.com/easybuilders/easybuild-easyconfigs/pull/22774#issuecomment-2904081080

It correctly installed the `easybuild` package

**Comment by boegel** [comment]
> successful report using the `EasyBuild-5.0.0.eb` easyconfig (with `easybuild` added to the sources) in: [easybuilders/easybuild-easyconfigs#22774 (comment)](https://github.com/easybuilders/easybuild-easyconfigs/pull/22774#issuecomment-2904081080)
> 
> It correctly installed the `easybuild` package

OK, but what if the `easybuild` package is not included in the `sources`, then it'll go down hard?

**Comment by branfosj** [comment]
> > successful report using the `EasyBuild-5.0.0.eb` easyconfig (with `easybuild` added to the sources) in: [easybuilders/easybuild-easyconfigs#22774 (comment)](https://github.com/easybuilders/easybuild-easyconfigs/pull/22774#issuecomment-2904081080)
> > It correctly installed the `easybuild` package
> 
> OK, but what if the `easybuild` package is not included in the `sources`, then it'll go down hard?

That works fine. This PR is required if it appears. If it does not appear then there is no change in behaviour.

@jfgrimm test report https://github.com/easybuilders/easybuild-easyblocks/pull/3700#issuecomment-2904065545 is that case. And I'll have one appear here soon for that case as well.

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)

Build succeeded for 0 out of 0 (2 easyconfigs in total)
bear-pg0208u17a - Linux RHEL 8.10, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/223277c0479039e0c2971844d174d327 for a full test report.

**Comment by jfgrimm** [comment]
the only edge case I can think of is that with #3725 merged, we might be overwriting the egg-info file in the case where `easbuild` _is_ in the sources

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-5.0.0.eb_ 
 * **SUCCESS** _EasyBuild-4.9.4.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0208u17a - Linux RHEL 8.10, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/f0f3695b09c12b27a53f4da90a5804e3 for a full test report.

**Comment by branfosj** [comment]
> the only edge case I can think of is that with #3725 merged, we might be overwriting the egg-info file in the case where `easbuild` _is_ in the sources

6aa115e to protect against that.

I'm happy for this to wait to after the release of 5.1.0.

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-5.0.0.eb_ 
 * **SUCCESS** _EasyBuild-4.9.4.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0208u17a - Linux RHEL 8.10, x86_64, AMD EPYC 9554 64-Core Processor (zen4), Python 3.6.8
See https://gist.github.com/branfosj/d55d27afae7f862a9d11e1694626af09 for a full test report.

**Comment by jfgrimm** [comment]
@branfosj I believe this is sufficiently guarded, so imo we should be fine to merge
I'll wait to see if @boegel objects

File: easybuild/easyblocks/e/easybuildmeta.py [hunk 2]
```diff
@@ -73,6 +73,10 @@ def __init__(self, *args, **kwargs):
             self.easybuild_pkgs.extend(['vsc-base', 'vsc-install'])
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
+        elif (LooseVersion(self.version) >= LooseVersion('5.0') and
+              any(f'easybuild-{self.version}' in source['filename'] for source in self.cfg['sources'])):
```
Comment (boegel) [inline easybuild/easyblocks/e/easybuildmeta.py:77]: `source` may not be a dict, depends on the easyconfig file being used, see `EasyBuild-4.9.1.eb` for example, so we need be be more careful here: https://github.com/branfosj/easybuild-easyblocks/pull/2

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **FAIL (unhandled exception: string indices must be integers)Traceback (most recent call last):
  File "/arcanine/scratch/gent/vo/000/gvo00002/vsc40023/easybuild/easybuild-framework/easybuild/main.py", line 160, in build_and_install_software
    (ec_res['success'], app_log, err_msg, err_code) = build_and_install_one(ec, init_env)
  File "/arcanine/scratch/gent/vo/000/gvo00002/vsc40023/easybuild/easybuild-framework/easybuild/framework/easyblock.py", line 5015, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/arcanine/scratch/gent/vo/000/gvo00002/vsc40023/easybuild/easybuild-framework/easybuild/framework/easyblock.py", line 4835, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/arcanine/scratch/gent/vo/000/gvo00002/vsc40023/easybuild/easybuild-framework/easybuild/framework/easyblock.py", line 4677, in run_step
    step_method(self)()
  File "/tmp/eb-8d442hzp/included-easyblocks-mp0_j1nj/easybuild/easyblocks/easybuildmeta.py", line 148, in install_step
    if not any(f'easybuild-{self.version}' in source['filename'] for source in self.cfg['sources']):
  File "/tmp/eb-8d442hzp/included-easyblocks-mp0_j1nj/easybuild/easyblocks/easybuildmeta.py", line 148, in <genexpr>
    if not any(f'easybuild-{self.version}' in source['filename'] for source in self.cfg['sources']):
TypeError: string indices must be integers
** _EasyBuild-4.9.1.eb_ 

Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3502.doduo.os - Linux RHEL 9.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.9.18
See https://gist.github.com/boegel/abd97e6f226207471a6dfeee279a43c9 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-5.0.0.eb_ 
 * **SUCCESS** _EasyBuild-4.9.1.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
node034.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/7f975722d3a5a7caa7103c4e48519b60 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-5.0.0_eb_in_src.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node034.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7643 48-Core Processor, Python 3.6.8
See https://gist.github.com/jfgrimm/5b7b1c122e6e45ff57cc260ae9e63fe6 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-5.0.0-eb_in_src.eb_ 
 * **SUCCESS** _EasyBuild-5.0.0.eb_ 
 * **SUCCESS** _EasyBuild-4.9.1.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0210u03a - Linux RHEL 8.10, x86_64, Intel(R) Xeon(R) Platinum 8480CL (sapphirerapids), Python 3.6.8
See https://gist.github.com/branfosj/d9d8fdd324944c04559200e568ad054c for a full test report.

**Comment by jfgrimm** [comment]
I've created https://github.com/easybuilders/easybuild-easyblocks/issues/3733 as a reminder to remove the egg-info workaround at some future point
I'd tag it as 6.0, but we don't have a milestone for that yet :P
